docs/HowToDevelopLoadableApps.md

```diff
@@ -0,0 +1,227 @@
+CONFIG_FLASH_PART_NAME="kernel,app,micom,micom,wifi,wifi,userfs,ftl,config,"
+```
+
+### Loading Applications into RAM
+
+- Sections of loadable apps will be allocated from user RAM region.
+- So find out the memory region large enough to load the appliacations.
+- Add its address & size to CONFIG_RAM_REGIONx_START & CONFIG_RAM_REGIONx_SIZE (if not already present).
+- While loading applications, memory manager (MM) uses this address space to
+  allocate application's text & data sections.
+
+  In below scenario, this region starts at 0x80200000 of size 30MB.
+
+```
+CONFIG_RAM_REGIONx_START="0x20200000,0x80200000"
+CONFIG_RAM_REGIONx_SIZE="262144,31457280"
+```
+
+## Add new loadable apps configuration
+
+## Configuration Requirements
+
+- Follow the requirements as stated here ([Configuration Settings](#configuration-settings)) to enable all configurations related to loadable apps.
+
+## Memory Requirements
+
+- Follow the requirements as stated here ([Memory Requirements](#memory-requirements)) for reserving flash & ram partitions for adding new applications.
+
+## Code Changes Required
+
+- Add required apps to "os/loadable_apps/loadable_sample/<app_name>"
+
+### Makefile
+
+- Make sure to generate elf type binary for each newly added application.
+  In <os/loadable_apps/loadable_sample/app/Makefile>
+
+```
+ifeq ($(CONFIG_APP_BINARY_SEPARATION),y)
+
+BIN = wifi
+BIN_TYPE = ELF
+BIN_VER = 20190412
+DYNAMIC_RAM_SIZE = 512000
+KERNEL_VER = 2.0
+STACKSIZE = 8192
+PRIORITY = 180
+
+include $(TOPDIR)/$(LOADABLEDIR)/loadable.mk
+endif # CONFIG_APP_BINARY_SEPARATION
+
+```
+Where,
+- BIN_TYPE: Binary Type (should be elf)
+- STACKSIZE: The first task will have a stack of this size
+- PRIORITY: The first (main) task of the app will be created with this priority
+- DYNAMIC_RAM_SIZE: This is the total amount of ram that will be available to the app. The developer must assess the max ram requirement for the app. During run time, if the memory requirement exceeds this value, then the app will fail. In that case the user might need to provide a higher value for this and rebuild the app and run it again.
+
+These values will be used by the kernel to start the first process of the application.
+
+### Make.defs
+
+- Include the app folder in this as follows;
+
+```
+CONFIGURED += loadable_sample/<app>
+```
+
+### Application code
+
+- Add binary manager notification in the main function of the app that has been added.
+
+```
+#ifdef CONFIG_BINARY_MANAGER
+        ret = binary_manager_notify_binary_started();
+        if (ret < 0) {
+                printf("MICOM notify 'START' state FAIL\n");
+        }
+#endif
+
+```
+- Below is the main signature for loadable apps where wifiapp being the non-critical app,
+  calls preapp_start() to load tash.
+
+```
+extern int preapp_start(int argc, char **argv);
+
+#ifdef CONFIG_APP_BINARY_SEPARATION
+int main(int argc, char **argv)
+#else
+int wifiapp_main(int argc, char **argv)
+#endif
+{
+.
+.
+.
```

Comment:
I think all the following points about tash, critical apps, and the test configs are not really necessary. 
It might provide too much information for adding a new app.

Reply:
makes sense, removed this information as of now